velox/exec/SortBuffer.cpp

```diff
@@ -139,6 +139,9 @@ void SortBuffer::noMoreInput() {
     auto spillPartition = spiller_->finishSpill();
     spillMerger_ = spillPartition.createOrderedReader();
   }
+
+  // Releases the unused memory reservation after procesing input.
+  pool_->release();
```

Comment:
QQ: Why don't we do pool_release() after `getOutput()`?

Reply:
As for now, we don't reserve any memory for output processing so it is better to release the extra memory hold by input and we could consider to add ensureOutputFit once per output buffer size.